pkg/repository/v1_repository.go

```diff
@@ -486,6 +499,38 @@ func (r *V1Repository) FetchComponent(item *v1manifest.VersionItem) (io.Reader,
 	return checkHash(reader, item.Hashes[v1manifest.SHA256])
 }
 
+// DownloadComponent downloads the component specified by item into local file,
+// the component will be removed if hash is not correct
+func (r *V1Repository) DownloadComponent(item *v1manifest.VersionItem, target string) error {
+	targetDir := filepath.Dir(target)
+	err := r.mirror.Download(item.URL, targetDir)
```

Comment:
seems we assume `item.URL` just a filename, but it may be  `/a/b/c/name`, but we have no such manifest yet
not introduce by this pr, seems the master code do this.
@lucklove PTAL